object_store/src/aws/client.rs

```diff
@@ -350,7 +350,7 @@ impl<'a> Request<'a> {
     }
 
     pub(crate) fn with_payload(mut self, payload: PutPayload) -> Self {
-        if !self.config.skip_signature || self.config.checksum.is_some() {
+        if self.config.sign_payload || self.config.checksum.is_some() {
```

Comment:
```suggestion
        if (!self.config.skip_signature && self.config.sign_payload) || self.config.checksum.is_some() {
```

We should compute the checksum if we are signing the request and payload signing is enabled

Reply:
Got bit by this too a while ago and my solution was the one @tustvold suggests

Reply:
Thanks! I applied your suggestion